src/main/java/com/jfrog/ide/idea/scan/ScannerFactory.java

```diff
@@ -0,0 +1,192 @@
+package com.jfrog.ide.idea.scan;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Maps;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.projectRoots.Sdk;
+import com.jetbrains.python.sdk.PythonSdkUtil;
+import com.jfrog.ide.common.scan.ScanLogic;
+import com.jfrog.ide.common.utils.PackageFileFinder;
+import com.jfrog.ide.idea.configuration.GlobalSettings;
+import com.jfrog.ide.idea.log.Logger;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+
+import static com.jfrog.ide.idea.scan.ScanUtils.createScanPaths;
+import static com.jfrog.ide.idea.utils.Utils.getProjectBasePath;
+
+/**
+ * Created by yahavi
+ */
+public class ScannerFactory {
+    private final Project project;
+
+    public ScannerFactory(Project project) {
+        this.project = project;
+    }
+
+    public static int getModuleIdentifier(String name, String path) {
+        return Objects.hashCode(name, path);
+    }
+
+    /**
+     * Scan projects, create new Scanners and delete unnecessary ones.
+     * Existing Scanners from previous scans, are not overridden.
+     */
+    public Map<Integer, ScannerBase> refreshScanners(Map<Integer, ScannerBase> oldScanners, ScanLogic scanLogic,
```

Comment:
I think we should consider renaming refreshScanners to create/initialize scanners.

Reply:
But if a scanner already exists, it's not created again.